trie/hash.go

```diff
@@ -0,0 +1,54 @@
+package trie
+
+import (
+	"encoding/hex"
+	"hash"
+
+	"golang.org/x/crypto/blake2s"
+)
+
+type hasher struct {
+	hash hash.Hash
+}
+
+func newHasher() (*hasher, error) {
+	key, err := hex.DecodeString("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f")
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := blake2s.New256(key)
+	if err != nil {
+		return nil, err
+	}
+
+	return &hasher{
+		hash: h,
+	}, nil
+}
+
+// Hash encodes the node and then hashes it if its encoded length is > 32 bytes
+func Hash(n node) (h []byte, err error) {
```

Comment:
Consider making this a receiver function, would allow optimizations further down the road

Reply:
done